mne/morph.py

```diff
@@ -0,0 +1,1443 @@
+        mri_to.affine,
+        mri_to.header.get_zooms()[:3],
+        spacing)
+
+    with warnings.catch_warnings(record=True):  # nibabel<->numpy warning
+        mri_to = nib.Nifti1Image(mri_to_res, mri_to_res_affine)
+
+    # get mri_to to world transform
+    mri_to_grid2world = mri_to.affine
+
+    # output mri_to as ndarray
+    mri_to = mri_to.dataobj[:, :, :]
+
+    # normalize values
+    mri_to = mri_to.astype('float') / mri_to.max()
+
+    # get mri_from to world transform
+    mri_from_grid2world = mri_from.affine
+
+    # output mri_from as ndarray
+    mri_from = mri_from.dataobj[:, :, :]
+
+    # normalize values
+    mri_from = mri_from.astype('float') / mri_from.max()
+
+    # compute center of mass
+    c_of_mass = imaffine.transform_centers_of_mass(mri_to, mri_to_grid2world,
+                                                   mri_from,
+                                                   mri_from_grid2world)
+
+    nbins = 32
+
+    # set up Affine Registration
+    affreg = imaffine.AffineRegistration(
+        metric=imaffine.MutualInformationMetric(nbins, None),
+        level_iters=list(niter_affine),
+        sigmas=[3.0, 1.0, 0.0],
+        factors=[4, 2, 1])
+
+    # translation
+    translation = affreg.optimize(mri_to, mri_from,
+                                  transforms.TranslationTransform3D(), None,
+                                  mri_to_grid2world, mri_from_grid2world,
+                                  starting_affine=c_of_mass.affine)
+
+    # rigid body transform (translation + rotation)
+    rigid = affreg.optimize(mri_to, mri_from,
+                            transforms.RigidTransform3D(), None,
+                            mri_to_grid2world, mri_from_grid2world,
+                            starting_affine=translation.affine)
+
+    # affine transform (translation + rotation + scaling)
+    affine = affreg.optimize(mri_to, mri_from,
+                             transforms.AffineTransform3D(), None,
+                             mri_to_grid2world, mri_from_grid2world,
+                             starting_affine=rigid.affine)
+
+    # apply affine transformation
+    mri_from_affine = affine.transform(mri_from)
+
+    # set up Symmetric Diffeomorphic Registration (metric, iterations)
+    sdr = imwarp.SymmetricDiffeomorphicRegistration(
+        metrics.CCMetric(3), list(niter_sdr))
+
+    # compute mapping
+    mapping = sdr.optimize(mri_to, mri_from_affine)
+    morph_params.update(
+        {'morph_shape': tuple(mapping.domain_shape.astype('float')),
+         'morph_zooms': spacing,
+         'morph_affine': mri_to_grid2world,
+         'AffineMap': affine.__dict__,
+         'DiffeomorphicMap': mapping.__dict__})
+
+    logger.info('done.')
+
+    return morph_params
+
+
+###############################################################################
+# Morph for SourceEstimate |  VectorSourceEstimate
+@deprecated("This function is deprecated and might be removed in a future "
```

Comment:
The deprecated functions need to be removed from `python_reference.rst`, no? Otherwise, new users will keep going to these old functions